src/sentry/lang/javascript/processor.py

```diff
@@ -1200,8 +1430,17 @@ def expand_frame(self, frame, source_context=None, source=None):
             return False
 
         if source_context is None:
-            # TODO: add lookup by debug id in case abs_path has it.
-            source = source or self.get_or_fetch_sourceview(url=frame["abs_path"])
+            # In this situation we didn't find the context in the source map, thus we need to load the original file.
+            # Here the logic for loading is either we load from the url or we try to fetch the file by debug id but only
+            # if it is a minified source.
```

Comment:
> but only if it is a minified source

you explicitly try to fetch a minified source, but how do we know that the call to `expand_frame` is actually being used for the minified source context?

Reply:
So, this logic is a bit tricky and non-explicit because it covers a lot of different code paths. `expand_frame` is called both for `new_frame` and `raw_frame`. We specify `MINIFIED_SOURCE` because by `debug_id` you can either index `SOURCE_MAP` or `MINIFIED_SOURCE`, otherwise, you fall back to `url` based search.

The high-level idea is if we have `debug_id` not None and we don't have a `source_context` it means that we didn't have the `sourcesContent` in the source map. If we don't have that context, we can't do much, therefore we try to resolve the frame by using the `MINIFIED_SOURCE`.

Reply:
A more explicit variant would actually be, if you fail to get the context and you have a debug_id, just try to fetch the minified_source in order to get at least the context from there. If you can't find the minified_source, it means that there is a problem with the debug_id system.